### DISCUSSION TITLE
Camera Usage in Seeed Studio XIAO ESP32S3 Sense | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#260] [Q&A] [answered]
# Camera Usage in Seeed Studio XIAO ESP32S3 Sense | Seeed Studio Wiki

This tutorial will explain how to properly use the camera on the XIAO ESP32S3 Sense.

https://wiki.seeedstudio.com/xiao_esp32s3_camera_usage/

**Comment by Mjrovai** [comment, marked as answer]
Some observations about the camera code on "Project I: Making a handheld camera": 
1. If you want to use the SD card reader available on the XIAO ESP32S Sense, you should change the SD.begin(D2) to SD.begin(21), so it is not necessary to cut off J3. 
2. The file saved on the SD card is not JPEG. It is raw data. It is necessary to convert the raw image to jpeg before saving. This is the code: 
  if(display_is_pressed()){
      Serial.println("display is touched");
      char filename[32];
      sprintf(filename, "/image%d.jpg", imageCount);
      
      // Save photo to file
      size_t out_len = 0;
      uint8_t* out_buf = NULL;
      esp_err_t ret = frame2jpg(fb, 12, &out_buf, &out_len);
      if (ret == false) {
        Serial.printf("JPEG conversion failed");
      } else {
        // Save photo to file
        writeFile(SD, filename, out_buf, out_len);
        Serial.printf("Saved picture: %s\n", filename);
        imageCount++;
        free(out_buf);
      }
    }

3. Reducing XCLK_FREQ_HZ from 20KHz to 10KHz prevent the warning message "EV-VSYNC-OVF" appearing on Serial Monitor.

**Reply by MatthewJeffson** [reply]
Wow! Thank you so much for sharing it!

**Comment by t-lie** [comment]
Hey,
today I received my S3 sense from seedstudio.
Is there any sample Code or is it possible to generate a H264 RTSP  stream for using within Frigate NVR ?
This would be great usecase.
thx

**Reply by MatthewJeffson** [reply]
Wow, it sounds a great idea!

**Comment by Peterhj05** [comment]
Hi. Can the OV5640 5MP Image Sensor?

**Comment by Peterhj05** [comment]
I was mean to ask, Can the OV5640 5MP Images sensor be connected instad of the supplied sensor?

**Reply by MatthewJeffson** [reply]
Yes, it can. The interesting thing is we are about launching the OV5640 image sensor :D
It is designed for (pin by pin) Seeed Studio XIAO ESP32S3 Sense.

**Reply by Peterhj05** [reply]
Hi Matthew. That is great news! Looking forward to hearing and seeing it!

**Reply by MatthewJeffson** [reply]
Hi, this is the link: https://www.seeedstudio.com/OV5640-Camera-for-XIAO-ESP32S3-Sense-With-Heat-Sink-p-5739.html
